pkg/apis/workspaces/v1alpha1/commands_conversion.go

```diff
@@ -40,38 +40,65 @@ func convertCommandTo_v1alpha2(src *Command, dest *v1alpha2.Command) error {
 	return nil
 }
 
-func convertCommandFrom_v1alpha2(src *v1alpha2.Command, dest *Command) error {
-	id := src.Key()
-	jsonCommand, err := json.Marshal(src)
-	if err != nil {
-		return err
-	}
-	err = json.Unmarshal(jsonCommand, dest)
-	if err != nil {
-		return err
+// getGroup returns the group the command belongs to
+func getGroup(dc v1alpha2.Command) *v1alpha2.CommandGroup {
+	switch {
+	case dc.Composite != nil:
+		return dc.Composite.Group
+	case dc.Exec != nil:
+		return dc.Exec.Group
+	case dc.Apply != nil:
+		return dc.Apply.Group
+	case dc.Custom != nil:
+		return dc.Custom.Group
+
+	default:
+		return nil
 	}
-	var destAttributes map[string]string
-	if src.Attributes != nil {
-		destAttributes = make(map[string]string)
-		err = convertAttributesFrom_v1alpha2(&src.Attributes, destAttributes)
+}
+
+func convertCommandFrom_v1alpha2(src *v1alpha2.Command, dest *Command) error {
+	if src != nil {
```

Comment:
I think it's better to reduce nesting here:
```go
if src == nil {
  return nil
}
// enclosed block

Reply:
updated!